### PR TITLE
IIO drivers header update

### PR DESCRIPTION
**Pull request by adrimbarean**
Move global variable definition from header files to sources in IIO drivers.

### CODE REVIEW

**Review by dbogdan — APPROVED** [review]

File: drivers/platform/aducm3029/iio_aducm3029.h [hunk 1]
```diff
@@ -113,13 +106,6 @@ enum iio_gpio_attributes {
...
 }
-static struct iio_attribute gpio_attributes[] = {
-	IIO_GPIO_ATTR("en", GPIO_ENABLE),
-	IIO_GPIO_ATTR("value", GPIO_VALUE),
-	IIO_GPIO_ATTR("is_output", GPIO_DIRECTION_OUTPUT),
-	IIO_GPIO_ATTR("number", GPIO_NUMBER),
-	END_ATTRIBUTES_ARRAY,
-};
 
 #define ADCUM3029_ADC_CH(idx)  {\
```
Comment (mchindri) [inline drivers/platform/aducm3029/iio_aducm3029.h:110]: You can move this too to .c file to be all in one place. These macros won't be used elsewhere

**Review by amiclaus — COMMENTED** [review]
There are some drivers build error on the latest modifications.

**Comment by adrimbarean** [comment]
V1:
 - fix astyle chackes;
 - fix driver build checks.

**Review by amiclaus — APPROVED** [review]

**Comment by adrimbarean** [comment]
V2:
 - move module only defines from header to source.

**Review by amiclaus — COMMENTED** [review]
New conflicts appeared after #934 merge.

**Comment by adrimbarean** [comment]
V3:
 - remove conflicting commits and will update them in separate pull requests.

**Pull request merged**